sentry/silkworm/sentry/common/ecc_key_pair.hpp

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2020-2022 The Silkworm Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+
+#include <string>
+#include <silkworm/common/base.hpp>
+
+namespace silkworm::sentry::common {
+
+class EccKeyPair {
+  public:
+    EccKeyPair();
+    explicit EccKeyPair(Bytes data);
+    explicit EccKeyPair(const ByteView& data);
```

Comment:
I suggest using `ByteView` instead of `const ByteView&` here for consistency at least within this code change (methods in `ecies_cipher` use pass-by-value).

This is just a hint, currently we have both pass-by-value and pass-by-ref for `ByteView` in different places within the codebase. However, IMO passing it by value is more idiomatic (see also https://quuxplusone.github.io/blog/2021/11/09/pass-string-view-by-value/)